Service/ProcessManager.php

```diff
@@ -90,15 +105,24 @@ public function newProcess(
             $process->setProcessType($processType);
         }
 
-        $process->setCaseNumber($this->getNewCaseNumber());
+        $resourceToLock = 'case-number';
+
+        $this->lockService->createLock($resourceToLock);
+        $this->lockService->acquire($resourceToLock, true);
```

Comment:
A comment telling that we're creating a blocking lock (https://symfony.com/doc/current/components/lock.html#blocking-locks) might be useful for future reference.